pkg/kube/kclient/client.go

```diff
@@ -0,0 +1,244 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kclient
```

Comment:
The name is a little confusing with pkg/kube/client.go

I think it is more of a controller framework, which wraps a kube client and register event handler

Reply:
My bad, i donot have a good idea either

Reply:
Ok, we can easily rename later if we come up with a better one. I'll think on it